File: frontend/src/app/game-list/models/game.ts
import { Developer } from "./developer"
import { Genre } from "./genre"
import { Platform } from "./platform"
import { Publisher } from "./publisher"
import { Review } from "./review"

export class Game {
  constructor(
    public id: number,
    public title: string,
    public summary: string,
    public description: string,
    public releaseDate: string,
    public rating: string,
    public price: number,
    public score: number,
    public publisher: Publisher,
    public developer: Developer,
    public platforms: Platform[],
    public genres: Genre[],
    public reviews: Review[]
  ) {}
}

File: backend/GameStoreAPI/Models/Game.cs
using System;
using System.Collections.Generic;

namespace GameStoreAPI.Models
{
    public class Game
    {
        public long Id { get; set; }
        public string Title { get; set; }
        public string Summary { get; set; }
        public string Description { get; set; }
        public DateTime ReleaseDate { get; set; }
        public string Rating { get; set; }
        public decimal Price { get; set; }
        public decimal Score { get; set; }

        public Publisher Publisher { get; set; }
        public long PublisherId { get; set; }
        public Developer Developer { get; set; }
        public long DeveloperId { get; set; }

        public IList<Platform> Platforms { get; set; }
        public IList<Genre> Genres { get; set; }
        public IList<Review> Reviews { get; set; }
    }
}
